### PR TITLE
Add Arrow/pandas support

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xhochy** [comment]
@mariusvniekerk FYI

**Comment by hajapy** [comment]
It seems they only build with extensions for OS X and windows. https://github.com/snowflakedb/snowflake-connector-python/blob/51d8dba6a3dcf5c60e087f023e5186e8b7175135/scripts/install.sh#L30
https://github.com/snowflakedb/snowflake-connector-python/blob/51d8dba6a3dcf5c60e087f023e5186e8b7175135/scripts/install.bat

**Comment by xhochy** [comment]
> Don’t forget to bump the build number.
> 
> We should also see if upstream project can be updated to support this without patches. The setup.py patch will likely need to be modified for all releases if past pattern holds (they update this file a lot).

I'll take care of taking this upstream once the build passes.

**Comment by xhochy** [comment]
Linux and OSX build failures are unrelated:

```
+ pip check
importlib-metadata 1.4.0 has requirement zipp>=0.5, but you have zipp 0.0.0.
```

**Comment by hajapy** [comment]
Maybe relevant: https://github.com/conda-forge/importlib-metadata-feedstock/issues/2
Some dependence might be bringing in importlib-metadata instead of importlib_metadata: 
https://github.com/conda-forge/importlib_metadata-feedstock/blob/master/recipe/meta.yaml

**Comment by xhochy** [comment]
> Maybe relevant: [conda-forge/importlib-metadata-feedstock#2](https://github.com/conda-forge/importlib-metadata-feedstock/issues/2)
> Some dependence might be bringing in importlib-metadata instead of importlib_metadata:
> https://github.com/conda-forge/importlib_metadata-feedstock/blob/master/recipe/meta.yaml

This is a step forward but won't help as the version in `zipp` is broken (and has been fixed in the meantime).

**Comment by hajapy** [comment]
linux_python2.7 failures seems to be due to sys.platform returning linux2 instead of just linux

windows failures are mostly because you can't set an env var on the script line in that way, however on py27 it fails due to an unsolvable conda environment.

osx py27 failure is a compile error (lots of warnings also) https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=113077&view=logs&j=9c5ef928-2cd6-52e5-dbe6-9d173a7d951b&t=113b264d-9551-5065-da10-7eb3bdeaf1bc&l=1470

So a possible simple path forward for building with extensions is to only support py36+ and run a different script line on windows.

**Comment by xhochy** [comment]
@conda-forge-admin please rerender

**Comment by xhochy** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ build:
   skip: true  # [py2k]
   script: 
     - "ENABLE_EXT_MODULES=true {{ PYTHON }} -m pip install . --no-deps -vvv"  # [unix]
-    - "set ENABLE_EXT_MODULES=true && call {{ PYTHON }} -m pip install . --no-deps -vvv"  # [win]
+    - "{ PYTHON }} -m pip install . --no-deps -vvv"  # [win]
```
Comment (hajapy) [inline recipe/meta.yaml]: missing a curly brace

**Comment by xhochy** [comment]
@hajapy Can you have another look here?

**Comment by hajapy** [comment]
Do you know why the windows builds can’t use the extensions?

**Comment by xhochy** [comment]
> Do you know why the windows builds can’t use the extensions?

Not really, this requires a bit more debugging and I would prefer to get the Linux/OSX builds in as the Windows debugging might take a bit longer.

**Comment by hajapy** [comment]
Ok, are there any more tests you can think to add that would help ensure things are working?

**Comment by xhochy** [comment]
The import test for `snowflake.connector.arrow_result` is the same as the Snowflake connector use internally to determine whether it should use the Arrow backend-

**Pull request merged**